src/orange/OrangeTrackView.hh

```diff
@@ -291,10 +357,7 @@ CELER_FUNCTION Propagation OrangeTrackView::find_next_step()
 
     if (!this->has_next_step())
     {
-        auto tracker = this->make_tracker(UniverseId{0});
-        auto isect = tracker.intersect(this->make_local_state());
-        next_step_ = isect.distance;
-        next_surface_ = isect.surface;
+        this->find_next_step_impl(no_intersection());
```

Comment:
Running `find_next_step_impl(no_intersection())` (effectively replacing `tracker.intersect(this->make_local_state());` with `tracker.intersect(this->make_local_state(), no_intersection());` means that the first method is no longer used in `SimpleUnitTracker`. Maybe replace with a templated method that forwards the arguments, so that the *first* `intersection` method gets the "infinite" vs "limited" restriction, and the remaining levels are searched with the maximum limit.